app/controllers/application_controller.rb

```diff
@@ -13,4 +14,25 @@ def configure_permitted_parameters
                                                         :slack,
                                                         :cohort ])
   end
+
+  private
+    def request_from_client?
+      !params[:client_id].nil?
+    end
+
+    def check_authenticity
+      if authentic_client_id?
+        store_current_location
+      else
+        head :forbidden
+      end
+    end
+
+    def store_current_location
```

Comment:
I'm going with `store_request_fullpath`. I'm also renaming `request_from_client?` to `client_request?`.